### PR TITLE
pretraining: add options to enable/disable lowercasing of input text

### PR DESCRIPTION
**Pull request by stefan-it**
Hi,

thanks for releasing the code and pre-trained models for ELECTRA :heart: 

In the original code base, the input text will be lowercased. However, for some tasks it is good to have a cased model.

This PR adds the option to enable or disable lowercasing of the input text.

More preciscely, the `--no_lower_case` option can be passed to the `build_openwebtext_pretraining_dataset.py` and `build_pretraining_dataset.py` to disable the lowercasing of text.

### CODE REVIEW

File: build_openwebtext_pretraining_dataset.py [hunk 1]
```diff
@@ -43,7 +43,8 @@ def log(*args):
       output_dir=os.path.join(args.data_dir, "pretrain_tfrecords"),
       max_seq_length=args.max_seq_length,
       num_jobs=args.num_processes,
-      blanks_separate_docs=False
+      blanks_separate_docs=False,
```
Comment (MeTaNoV) [inline build_openwebtext_pretraining_dataset.py]: The doc says True by default.

File: build_openwebtext_pretraining_dataset.py [hunk 2]
```diff
@@ -78,6 +79,11 @@ def main():
                       help="Number of tokens per example.")
   parser.add_argument("--num-processes", default=1, type=int,
                       help="Parallelize across multiple processes.")
+  parser.add_argument("--do-lower-case", dest='do_lower_case',
```
Comment (MeTaNoV) [inline build_openwebtext_pretraining_dataset.py]: no add_argument for blanks_separate?

Comment (stefan-it) [reply build_openwebtext_pretraining_dataset.py]: Thanks for that hint! I added it with the last commit and set it to `False` like it is used in the  OpenWebTextCorpus script.

File: build_openwebtext_pretraining_dataset.py (hunk 1 above)
Comment (stefan-it) [reply build_openwebtext_pretraining_dataset.py]: By now it is only `False` for the  OpenWebTextCorpus script. The main readme mentions the more generic `build_pretraining_dataset.py` script, where it is set to `True` :)

File: build_openwebtext_pretraining_dataset.py (hunk 2 above)
Comment (clarkkev) [reply build_openwebtext_pretraining_dataset.py]: I actually don't think it is needed, as you should always have blanks-separate-docs=False for the OpenWebTextCorpus (each doc is a separate file and there is a blank between every sentence).

Comment (stefan-it) [reply build_openwebtext_pretraining_dataset.py]: No problem, I just reverted the last commit :)

**Pull request merged**